pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1419,40 +1428,38 @@ private ByteBuf processMessageChunk(ByteBuf compressedPayload, MessageMetadata m
      * Notify waiting asyncReceive request with the received message.
      *
      * @param message
+     * @return the message is enqueue incomingMessages.
      */
-    void notifyPendingReceivedCallback(final Message<T> message, Exception exception) {
-        if (pendingReceives.isEmpty()) {
-            return;
-        }
-
+    boolean notifyPendingReceivedCallback(final Message<T> message, Exception exception) {
         // fetch receivedCallback from queue
         final CompletableFuture<Message<T>> receivedFuture = nextPendingReceive();
         if (receivedFuture == null) {
-            return;
+            return message != null && getCurrentReceiverQueueSize() != 0;
```

Comment:
Can we just return true here?

Reply:
Maybe not. if CurrentReceiverQueueSize == 0, the message can't enqueue incoming msg.